### PR TITLE
[MM-10768] Migrate "Channel.SaveDirectChannel" to Sync by default

### PR DESCRIPTION
**Pull request by iomodo**
#### Summary
This PR will Migrate `Channel.SaveDirectChannel` and `Channel.CreateDirectChannel` to sync

#### Ticket Link
https://github.com/mattermost/mattermost-server/issues/10768

### CODE REVIEW

File: store/sqlstore/channel_store.go [hunk 1]
```diff
@@ -510,64 +510,62 @@ func (s SqlChannelStore) CreateDirectChannel(userId string, otherUserId string)
...
 
-		if err := transaction.Commit(); err != nil {
-			result.Err = model.NewAppError("SqlChannelStore.SaveDirectChannel", "store.sql_channel.save_direct_channel.commit.app_error", nil, err.Error(), http.StatusInternalServerError)
-			return
-		}
+	if err := transaction.Commit(); err != nil {
+		return nil, model.NewAppError("SqlChannelStore.SaveDirectChannel", "store.sql_channel.save_direct_channel.commit.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	return newChannel, apperr
```
Comment (jespino) [inline store/sqlstore/channel_store.go]: Here `apperr` must be nil, so, is way clearer to just write `return newChannel, nil`

File: store/storetest/channel_store.go [hunk 2]
```diff
@@ -394,7 +394,10 @@ func testChannelStoreGet(t *testing.T, ss store.Store, s SqlSupplier) {
 	m2.UserId = u2.Id
 	m2.NotifyProps = model.GetDefaultChannelNotifyProps()
 
-	store.Must(ss.Channel().SaveDirectChannel(&o2, &m1, &m2))
+	if _, err := ss.Channel().SaveDirectChannel(&o2, &m1, &m2); err != nil {
```
Comment (jespino) [inline store/storetest/channel_store.go]: We prefer to user here something like:
```go
_, err := ss.Channel().SaveDirectChannel(&o2, &m1, &m2)
require.Nil(t, err)
```

Comment (jespino) [reply store/storetest/channel_store.go]: And of course the same for all the other cases where you just put a `panic` there.

Comment (iomodo) [reply store/storetest/channel_store.go]: Done. 
@jespino, thanks for the review :)

File: store/sqlstore/channel_store.go [hunk 3]
```diff
@@ -510,64 +510,62 @@ func (s SqlChannelStore) CreateDirectChannel(userId string, otherUserId string)
...
+
+	directchannel.TeamId = ""
+	// After updating saveChannelT() should be:
+	// newChannel, apperr := s.saveChannelT(transaction, directchannel, 0)
+	channelResult := s.saveChannelT(transaction, directchannel, 0)
+	var newChannel *model.Channel
+	if channelResult.Data != nil {
+		newChannel = channelResult.Data.(*model.Channel)
+	}
+	apperr := channelResult.Err
```
Comment (lieut-data) [inline store/sqlstore/channel_store.go]: Can we name this `appErr` for consistency with other parts of the code?

File: store/storetest/channel_store.go [hunk 4]
```diff
@@ -2494,8 +2497,10 @@ func testChannelStoreAutocompleteInTeamForSearch(t *testing.T, ss store.Store, s
 	o5.Type = model.CHANNEL_PRIVATE
 	store.Must(ss.Channel().Save(&o5, -1))
 
-	store.Must(ss.Channel().CreateDirectChannel(u1.Id, u2.Id))
-	store.Must(ss.Channel().CreateDirectChannel(u2.Id, u3.Id))
+	_, err := ss.Channel().CreateDirectChannel(u1.Id, u2.Id)
```
Comment (hanzei) [inline store/storetest/channel_store.go]: This fixes the build. You also have the fix the same error in L2639
```suggestion
	_, err = ss.Channel().CreateDirectChannel(u1.Id, u2.Id)
```

**Pull request merged**